apps/api/src/bank-transactions-file/bank-transactions-file.controller.ts

```diff
@@ -75,32 +76,47 @@ export class BankTransactionsFileController {
         )
       }),
     )
-
     //now import the parsed donations
-    const donations: CreateManyBankPaymentsDto[] = []
-    for (const fileOfBankMovements of allMovementsFromAllFiles) {
-      for await (const movement of fileOfBankMovements) {
-        const campaign = await this.campaignService.getCampaignByPaymentReference(
-          movement.paymentRef,
-        )
+    const bankImportRecords: BankImport[] = []
+    for (const movement of allMovementsFromAllFiles) {
+      const campaign = await this.campaignService.getCampaignByPaymentReference(movement.paymentRef)
+      const donationImportRecord: BankImport = {
+        status: ImportStatus.UNPROCESSED,
+        amount: movement.payment.amount,
+        currency: movement.payment.currency,
+        createdAt: movement.payment.createdAt,
+        extPaymentIntentId: movement.payment.extPaymentIntentId,
+      }
 
-        if (!campaign) {
-          Logger.warn('No campaign with payment reference: ' + movement.paymentRef)
-          throw new NotFoundException('No campaign with payment reference: ' + movement.paymentRef)
-        }
+      if (!campaign) {
+        const errorMsg = 'No campaign with payment reference: ' + movement.paymentRef
+        donationImportRecord.status = ImportStatus.FAILED
+        donationImportRecord.message = errorMsg
+        Logger.warn(errorMsg)
+        bankImportRecords.push(donationImportRecord)
+        continue
+      }
+
+      const vault = await this.vaultService.findByCampaignId(campaign.id)
+      movement.payment.extPaymentMethodId = 'imported bank payment'
+      movement.payment.targetVaultId = vault[0].id
+      movement.payment.type = DonationType.donation
+      movement.payment.status = DonationStatus.succeeded
+      movement.payment.provider = PaymentProvider.bank
 
-        const vault = await this.vaultService.findByCampaignId(campaign.id)
-        movement.payment.extPaymentMethodId = 'imported bank payment'
-        movement.payment.targetVaultId = vault[0].id
-        movement.payment.type = DonationType.donation
-        movement.payment.status = DonationStatus.succeeded
-        movement.payment.provider = PaymentProvider.bank
-        donations.push(movement.payment)
+      try {
+        donationImportRecord.status = await this.donationsService.createUpdateBankPayment(
+          movement.payment,
+        )
```

Comment:
well done with the refactoring of processing the donations in the one cycle  - the previous code was more complex and creating additional collection that was cycled separately